### ISSUE TITLE
ClassCastException in CreateKeyFinalFragment.java:190

### ISSUE THREAD

**Issue by dschuermann** [open-keychain/open-keychain#1005]
from G Play

```
java.lang.ClassCastException: org.sufficientlysecure.keychain.operations.results.EditKeyResult cannot be cast to org.sufficientlysecure.keychain.operations.results.SaveKeyringResult
at org.sufficientlysecure.keychain.ui.CreateKeyFinalFragment$4.handleMessage(CreateKeyFinalFragment.java:190)
at android.os.Handler.dispatchMessage(Handler.java:102)
at android.os.Looper.loop(Looper.java:136)
at android.app.ActivityThread.main(ActivityThread.java:5105)
at java.lang.reflect.Method.invokeNative(Native Method)
at java.lang.reflect.Method.invoke(Method.java:515)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:792)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:608)
at dalvik.system.NativeStart.main(Native Method)
```

**Comment by Valodim** [comment]
this looks like yet another problem which can be fixed by getting rid of logic in KeychainIntentService. working on it.

**Comment by Valodim** [comment]
Should be fixed in 320f7d35efb059b99b31506426554e9a8f138d8f
